dashboard/app/helpers/users_helper.rb

```diff
@@ -199,99 +199,95 @@ def script_progress_for_users(users, script)
+            level, user, user_levels_by_level, sl, paired_user_levels, include_timestamp
+          )
+          if bubble_choice_progress
+            progress.merge!(bubble_choice_progress.compact)
+          end
           next
         end
 
         contained_level_id = level.contained_levels.try(:first).try(:id)
         ul = user_levels_by_level.try(:[], contained_level_id || level_id)
-        level_progress = get_level_progress(ul, sl, paired_user_levels, include_timestamp)
-
-        if level_progress[:status] == LEVEL_STATUS.not_tried
-          # for now, we don't allow authorized teachers to be "locked"
-          if level_progress[:locked] && !user.authorized_teacher?
-            progress[level_id] = {
-              # TODO: Stop sending status here when front-end stops using LEVEL_STATUS.locked (LP-1865)
-              status: LEVEL_STATUS.locked,
-              locked: true
-            }
-          end
-          next
-        end
+        level_progress = get_level_progress(
+          ul, sl, paired_user_levels, include_timestamp
+        )
 
-        progress[level_id] = level_progress
+        next unless level_progress
 
-        # Just in case this level has multiple pages, in which case we add an additional
+        # if the level has multiple pages, we add an additional
         # array of booleans indicating which pages have been completed.
-        pages_completed = get_pages_completed(user, sl)
-
-        next unless pages_completed
+        level_progress[:pages_completed] = get_pages_completed(user, sl)
 
-        progress[level_id][:pages_completed] = pages_completed
-        pages_completed.each_with_index do |result, index|
-          progress["#{level_id}_#{index}"] = {
-            result: result,
-            submitted: level_progress[:submitted],
-            readonly_answers: level_progress[:readonly_answers]
-          }.compact
-        end
+        progress[level_id] = level_progress.compact
       end
     end
     progress
   end
 
   # Summarizes a user's progress on a particular level
-  private def get_level_progress(user_level, script_level, paired_user_levels, include_timestamp)
-    completion_status = activity_css_class(user_level)
-    locked = user_level&.show_as_locked?(script_level.lesson) || script_level.lesson.lockable? && !user_level
-
-    # For levels that are not tried we will either return no progress or a locked status for locked levels.
-    # In either case, we do not need a full set of progress data so we return early here.
-    if completion_status == LEVEL_STATUS.not_tried
-      return {
-        status: completion_status,
-        locked: locked || nil
-      }
+  private def get_level_progress(
+    user_level,
+    script_level,
+    paired_user_levels,
+    include_timestamp
+  )
+    # if we don't have a user level, we don't need to return anything,
+    # unless the lesson is lockable, in which case that means it's locked
```

Comment:
Could you rephrase this comment?   It's a little bit difficult to parse.  (It might help to include _why_ we want this behavior.